rust/lance-encoding/src/encodings/logical/struct.rs

```diff
@@ -308,6 +351,18 @@ impl LogicalPageScheduler for SimpleStructScheduler {
             );
             rows_to_read -= min_rows_added;
             current_top_level_row += min_rows_added as u64;
+            if self.is_root {
+                trace!(
+                    "Scheduler scan complete ({} rows now scheduled)",
+                    current_top_level_row - top_level_row
+                );
+                context
+                    .sink
```

Comment:
nit: should we have a similar method on context like `emit` to send a ScanLine to the sink?

Reply:
I kind of intentionally hid this because encodings shouldn't be able to send scan lines.  The struct encoding is a "special" encoding because we will always have one struct encoding that lives at the top of the decoder tree (even if there is only one column).  In the next PR I make this even more obvious.  So, it is somewhat intentional here that the struct encoding has to peek into the private impl of the scheduler context.